src/data/markdown/translated-guides/en/03 Results output/100 End-of-test/150-custom-summary.md

```diff
@@ -1,35 +1,76 @@
 ---
 title: Custom summary
-excerpt: With the handle summary, you can customize every part of your report, 
+excerpt: With the handlesummary() function, you can customize every part of your report. Change the content, redirect output, and more.
 ---
 
-Use the `handleSummary()` function to completely customize the end-of-test summary report.
-
-k6 calls`handleSummary()` at the end of the test run, even after [`teardown()`](/using-k6/test-life-cycle).
-If `handleSummary()` is exported, k6 does _not_ print the default summary.
-
-Besides customizing the CLI summary, you can also transform the summary data into machine- or human-readable formats.
-This lets you create JS-helper functions that generate JSON, CSV, XML (JUnit/xUnit/etc.), HTML, etc. files from the summary data.
+With the `handleSummary()` callback, you can completely customize your end-of-test summary.
+In this document, you can read about:
+- How `handleSummary()` works
+- How to customize the content and output location of your summary
+- The data structure of the summary object
 
 <Blockquote mod="note"
-title="handleSummary() is available on only local tests."
->
+title="handleSummary() is available only for local tests.">
 
-However we plan to support the feature for k6 Cloud tests, too.
+However, we plan to support the feature for k6 Cloud tests, too.
 [Track progress in this issue](https://github.com/grafana/k6-cloud-feature-requests/issues/24).
 
 </Blockquote>
 
-## Data format returned by `handleSummary()`
+## About the `handleSummary()` callback
+
+After your VU code runs, k6 aggregates your metrics into an object.
```

Comment:
We mostly talk about test results and end-of-test summary. The word `object` does not sound familiar at this moment. 

One attempt:

> After k6 executes the load test, it aggregates the test results into a summary. The handleSummary() function receives this summary as an argument (called data in all examples here). 

> You can use handleSummary() to manipulate the test result summary as you want. To get an idea of what the data looks like, run this script and open the output file, summary.json.

Reply:
I am also against the wording of

> VU code runs

IMO this has always been confusing, but IMO in this case can just use "the load test has been executed" or some other variant of the "test was executed" instead of specyfing some code that has run and name it something,

Reply:
Maybe I'm wrong, but to me this sentence is a key detail:

> Fundamentally, `handleSummary()` is just a function that can access a data object.

So, I'm mildly against removing the reference to `object` in other places because:
- It causes the doc to lose coherence (unless we remove all mentions of the data object).
- It's just less specific, calling it a "summary" doesn't describe the data as well as calling it a "summary object". And working with objects is a fundamental aspect of JavaScript programming, so I don't think it adds new vocabulary or unnecessary complication. Am i making an incorrect assumption here?


On the other hand, I agree with removing the reference to VU code and to callbacks, because these internal details don't really inform how a developer would code a `handleSummary()`.

Reply:
> So, I'm mildly against removing the reference to object in other places because:

I am also against not calling it `object` or `data` :thinking:. Object is bit vague in general ... so maybe `JavaScript object` just to make certain people understand that in this case we do mean a simpla javascript object.

> You can use handleSummary() to manipulate the test result summary as you want. 

IMO "manimulate" is the wrong word. You return the summary you want, you don't manipulate anything. In practice you can return w/e you want even something completely unrelated to the summary object provided.